src/silicone/multiple_infillers/split_collection_with_remainder_emissions.py

```diff
@@ -0,0 +1,196 @@
+"""
+Uses the 'quantile rolling windows' database cruncher to infill most constituents and
+infills the remainder as another specified emissions type (which may be negative).
+"""
+
+import logging
+
+from silicone.database_crunchers import QuantileRollingWindows
+from silicone.multiple_infillers import infill_composite_values
+from silicone.utils import _remove_equivs, convert_units_to_MtCO2_equiv
+
+logger = logging.getLogger(__name__)
+
+
+class SplitCollectionWithRemainderEmissions:
+    """
+    Splits the known aggregate emissions into several components with the 'quantile
+    rolling windows' cruncher, then sets allocates any the remainder to 'remainder'.
+    """
+
+    def __init__(self, db):
+        """
+        Initialises the database to use for infilling.
+
+        Parameters
+        ----------
+        db : IamDataFrame
+            The database for infilling.
+        """
+        self._db = db.copy()
+
+    def _check_and_return_desired_unit(
+        self, relevant_df, aggregate, components, remainder
+    ):
+        """
+            Converts the units of the component emissions to be the same as the
+            aggregate emissions. Returns the converted database and the unit.
+
+            Parameters
+            ----------
+            relevant_df : :obj:`pyam.IamDataFrame`
+                Data with units that need correcting.
+
+            aggregate : str
+                The name of the aggregate variable.
+
+            components : [str]
+                List of the names of the variables to be summed.
+
+            remainder : str
+                The component which will be constructed as a remainder.
+
+            Return
+            ------
+            :obj:`pyam.IamDataFrame`
+                Data with consistent units.
+
+            str
+                The unit of the aggregate data.
+
+            Raises
+            ------
+            ValueError:
+            The variables in this dataframe have units that cannot easily be converted
+            to make them consistent.
+            """
+        all_var = [aggregate, remainder] + components
+        all_units = relevant_df.variables(True)
+        if not all(var in all_units["variable"].values for var in all_var):
+            "Some variables missing from database when performing unit " "conversion: {}".format(
```

Comment:
is this meant to be an error message or print statement?

Reply:
A warning actually (fixed now)